src/openzeppelin/utils/udc.cairo

```diff
@@ -0,0 +1,40 @@
+#[contract]
+mod UniversalDeployerContract {
+    use array::ArrayTrait;
+    use hash::pedersen;
+    use starknet::{class_hash::ClassHash, ContractAddress, get_caller_address};
+    use traits::Into;
+
+    #[event]
+    fn ContractDeployed(
+        address: ContractAddress,
+        deployer: ContractAddress,
+        unique: bool,
+        class_hash: ClassHash,
+        calldata: Array<felt252>,
+        salt: felt252,
+    ) {}
+
+    #[external]
+    fn deploy_contract(
+        class_hash: ClassHash, salt: felt252, unique: bool, calldata: Array<felt252>
```

Comment:
```suggestion
        class_hash: ClassHash, salt: felt252, unique: bool, calldata: Span<felt252>
```
should we use `Span` instead of `Array`, considering that's what the syscall takes? we don't mean to own/modify it anyway

Reply:
Yeah, `span` is the better choice